### ISSUE TITLE
An option ForceIf for Style/GuardClause cop or a new cop that prevents use of unless for complex conditions

### ISSUE THREAD

**Issue by vittorius** [rubocop/rubocop#10828]
## Is your feature request related to a problem? Please describe.

Guard clauses are poorly readable when applied to `unless` conditions. It would be great to have an option for this cop that "turns off" the Style/NegatedIf cop regardless of its actual setting or presence (takes a priority over it).

## Describe the solution you'd like

I'd like to suggest an option `ForceIf` what will force `if` usage only for the guard clauses

```yaml
Style/GuardClause:
  ForceIf: true
```

E.g.

```ruby
return unless start_year.present? && two_digit_end_year.present?

generate_date!
```

could be

```
generate_date! if start_year.blank? || two_digit_end_year.blank?
```

## Describe alternatives you've considered

I've searched through all of the existing cops but nothing came to my mind in regards to possible alternatives.
But, I was thinking about a different cop that will forbid using `unless` with complex conditions (more than 1 term):

```ruby
# bad
return unless user.present? && Account.exists?(user: user)

# good
return unless user.present?
```

**Comment by dvandersluis** [comment]
`Style/UnlessLogicalOperators` with `EnforcedStyle: forbid_logical_operators` should do what you want here. It is not enabled by default so you'd have to enable it in your config.

```ruby
Style/UnlessLogicalOperators:
  Enabled: true
  EnforcedStyle: forbid_logical_operators
```

**Comment by vittorius** [comment]
Huge thanks!
